wagtailnews/views/frontend.py

```diff
@@ -9,13 +9,13 @@
 
 def _newsitem_list(request, newsindex, newsitem_list, extra_context):
     paginator, page = paginate(request, newsitem_list)
-    context = {
-        'self': newsindex,
+    context = newsindex.get_context(request)
+    context.update({
         'page': newsindex,
```

Comment:
`page` will also come from `newsindex.get_context(request)`, so you can remove that line as well.